### DISCUSSION TITLE
Change TO email address if you are in development mode

### DISCUSSION THREAD

**Discussion by beolafsen** [oqtane/oqtane.framework#2121] [Ideas] [answered]
Will it be possible to enter a check / test in NotificationJob.cs where you change the email address you send to if the system is in development mode? 
Finds it very useful when I have to test new functions or look for errors in existing systems. I can then copy the database from a customer and then enter an alternative email address (eg in settings) so that all emails are sent to me instead of to the email addresses listed on the customers.

What needs to be done:
1) Create new parameter under site - ex:-> TestEmailAdr = "my@email.com"
2) In NotificationJob.cs
`if (env.IsDevelopment () && settings ["TestmailAdr"] ! = "")
  {
      notification.Body = notification.ToEmail + "" + notification.Body;
      notification.ToEmail = settings ["TestmailAdr"];
  }``
In this way we can also see who the original email-adr.

Bst regards
Bjørn-Even

**Comment by sbwalker** [comment, marked as answer]
@beolafsen I understand your use case however I would prefer to not include this feature into the framework itself as we are trying to avoid a lot of obscure configuration switches which can become difficult to maintain in the long term. An alternative approach which would work for you is to simply create your own Notification job (based on the code in the existing job) and use it in your own environments. This is the benefit of the extensibility model in Oqtane.

**Reply by beolafsen** [reply]
Ok - good alternativ soulution.

**Comment by beolafsen** [comment]
Would it be possible to send EnvironmentName together with the rest of the system info parameters?
I can also make my own service/corntroller for this - but if other also would have use for this it can be solved in the service/controller that allready exist.

 `   systeminfo.Add("EnvironmentName", _environment.EnvironmentName);`

**Comment by sbwalker** [comment]
@beolafsen yes it can be added... however keep in mind that the API for retrieving this information is restricted to host users only, as access to system info needs to be restricted. So depending on the type of client that you are creating to access this API, you will need to verify that the user will have the necessary permissions.

**Reply by beolafsen** [reply]
Good point - I will probably need that information for non-host users as well.
 It seams that I have to create my own API.

BEO
